Content.Server/Station/Systems/StationJobsSystem.cs

```diff
@@ -206,8 +208,16 @@ public bool TryRemovePlayerJobs(EntityUid station,
     {
         if (!Resolve(station, ref jobsComponent, false))
             return false;
-
-        return jobsComponent.PlayerJobs.Remove(userId);
+        // DeltaV added RemovedPlayerJobsEvent for CaptainStateSystem
+        if (jobsComponent.PlayerJobs.TryGetValue(userId, out var playerJobsEntry))
+        {
+            jobsComponent.PlayerJobs.Remove(userId);
```

Comment:
this can just be a .Remove in place of .TryGetValue

Reply:
I need the value for playerJobsEntry for the event.

Reply:
`.Remove` does two things in a single atomic operation: if it finds the key, it both removes the key-value pair from the dictionary AND copies the value into the out parameter. So you can use it instead of first retrieving the value and them removing it.

Reply:
oh i see nice

Reply:
i didn't realize it had an out